python/katana/local/_graph.pyx

```diff
@@ -457,22 +460,72 @@ cdef class GraphBase:
 
         :rtype: list[EntityType]
         """
-        cdef const EntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetNodeTypeManager()
         type_ids = manager.GetAtomicEntityTypeIDs()
         types = [EntityType.make(manager, type_id) for type_id in type_ids]
         return types
 
+    def get_node_type_manager(self):
+        """
+        :return the node type manager
+        """
+        return EntityTypeManager.make(&self.underlying_property_graph().GetNodeTypeManager())
+
+    def get_type_of_node(self, uint64_t n):
+        """
+        Return the type id of a node
+
+        :param n: node id
+        :return the type id of the node
+        """
+        return self.underlying_property_graph().GetTypeOfNode(n)
+
+    def does_node_have_type(self, uint64_t n, EntityTypeID type_id):
+        """
+        Check whether a given node has a certain type
+
+        :param n: node id
+        :param type_id: type id
+        :return True if node n has the given type and False otherwise
+        """
+        return self.underlying_property_graph().DoesNodeHaveType(n, type_id)
+
     @property
     def edge_types(self):
         """
         The types of atomic edge types in the graph.
 
         :rtype: list
         """
-        cdef const EntityTypeManager* manager = &self.underlying_property_graph().GetEdgeTypeManager()
+        cdef const CEntityTypeManager* manager = &self.underlying_property_graph().GetEdgeTypeManager()
         types = manager.GetAtomicEntityTypeIDs()
         return [EntityType.make(manager, typeid) for typeid in types]
 
+    def get_edge_type_manager(self):
+        """
+        :return the edge type manager
+        """
+        return EntityTypeManager.make(&self.underlying_property_graph().GetEdgeTypeManager())
+
+    def get_type_of_edge(self, uint64_t e):
+        """
+        Return the type id of an edge
+
+        :param e: edge id
+        :return the type id of the edge
+        """
+        return self.underlying_property_graph().GetTypeOfEdge(e)
+
+    def does_edge_have_type(self, uint64_t e, EntityTypeID type_id):
+        """
+        Check whether a given edge has a certain type
+
+        :param e: edge id
+        :param type_id: type id
+        :return True if edge e has the given type and False otherwise
+        """
```

Comment:
This is OK for the moment. But I think we should move all this to the edge dataframe. That way people don't need to handle edge IDs as much.

Please add a TODO and file a JIRA issue so we don't forget to come back and improve this.